addons/test_mail_full/tests/test_rating.py

```diff
@@ -169,10 +171,69 @@ def test_rating_last_value_perfs(self):
 class TestRatingRoutes(HttpCase, TestRatingCommon):
 
     def test_open_rating_route(self):
+        """
+        _summary_
+
+        With recent changes the expected behavior:
+        1) Clicking on the smiley image triggers the /rate/<string:token>/<int:rate>
+        route. SHOULD NOT update the rating of the record but only return the
+        feedback FORM website page
+        2) Customer interacts with webpage and submits FORM. Triggers /rate/<string:token>/submit_feedback
+        route. Should update the rating of the record with the data in the POST request
+        """
+        self.authenticate(None, None)  # set up session for public user
         access_token = self.record_rating._rating_get_access_token()
-        self.url_open(f"/rate/{access_token}/5")
 
+        # First round of clicking the URL and then submitting FORM data
+
+        response_click_one = self.url_open(f"/rate/{access_token}/5")
+        response_click_one.raise_for_status()
+
+        # check returned form
+        self.assertEqual(response_click_one.headers['Content-Type'], 'text/html; charset=utf-8')
+        forms = lxml.html.fromstring(response_click_one.content).xpath('//form')
+        self.assertEqual(forms[0].get('method'), 'post')
+        self.assertEqual(forms[0].get('action', ''), f'/rate/{access_token}/submit_feedback')
+
+        # rating should not change, i.e. default values
         rating = self.record_rating.rating_ids
-        self.assertTrue(rating.consumed)
-        self.assertEqual(rating.rating, 5)
+        self.assertFalse(rating.consumed)
+        self.assertEqual(rating.rating, 0)
+        self.assertFalse(rating.feedback)
+        self.assertEqual(self.record_rating.rating_last_value, 0)
+
+        response_submit_one = self.url_open(f"/rate/{access_token}/submit_feedback",
+                                        data={"rate": 5,
+                                              "csrf_token": http.Request.csrf_token(self),
+                                              "feedback": "bad"})
```

Comment:
Isn't 5 technically "good" :D ?

Reply:
You are right (will fix int in next squash). /rate/tde/5